<launch>
  <!-- 相机驱动部分 -->
  <group ns="usb_cam">
    <node name="hardware" pkg="nodelet" type="nodelet" 
          args="standalone usb_cam_hardware/USBCamHardware" output="screen">
      <remap from="camera_info_controller/camera_info" to="camera_info"/>
      <remap from="packet_controller/image" to="packet"/>
      <remap from="yuyv_controller/image" to="image_raw"/>
      <param name="video_device" value="/dev/video4"/>
      <param name="image_width" value="640"/>
      <param name="image_height" value="480"/>
      <param name="pixel_format" value="yuyv"/>
      <param name="framerate" value="30"/>
    </node>

    <node name="controller_starter" pkg="controller_manager" type="controller_manager" 
          args="spawn camera_info_controller packet_controller yuyv_controller" output="screen"/>

    <!-- 控制器参数 -->
    <param name="camera_info_controller/type" value="usb_cam_controllers/CameraInfoController"/>
    <param name="camera_info_controller/camera_frame_id" value="baffalo_uvc_camera"/>

    <param name="packet_controller/type" value="usb_cam_controllers/PacketController"/>
    <param name="packet_controller/format" value="yuv422"/>

    <param name="yuyv_controller/type" value="usb_cam_controllers/YUYVController"/>
    <param name="yuyv_controller/image_width" value="640"/>
    <param name="yuyv_controller/image_height" value="480"/>
  </group>

  <!-- YOLO检测节点 -->
  <node pkg="yolo_ros" type="yolo_ros.py" name="yolo_ros" output="screen">
    <!-- 参数加载 -->
    <rosparam command="load" file="$(find yolo_ros)/config/camera_params.yaml"/>
    
    <!-- 基础参数 -->
    <param name="model_path" value="/root/model/basketballn_openvino_model_fp32"/>
    <param name="conf_threshold" value="0.6"/>
    <param name="iou_threshold" value="0.45"/>
    <param name="enable_undistort" value="true"/>
    <!-- 性能参数 -->
    <param name="queue_size" value="1"/>
    <param name="num_threads" value="1"/>
  </node>
</launch>